src/main/java/ca/uhn/fhir/jpa/starter/CollectDataResourceProvider.java

```diff
@@ -0,0 +1,155 @@
+        MeasureReport report = createMeasureReport(fullUrl);
+        Bundle observationBundle = createObsBundle(fullUrl);
+  
+        parameters.addParameter(new Parameters.ParametersParameterComponent()
+                .setName(PlirConstants.PARAMETER_NAME_MEASURE_REPORT).setResource(report));
+
+        parameters.addParameter(new Parameters.ParametersParameterComponent()
+                .setName(PlirConstants.PARAMETER_NAME_RESOURCE).setResource(observationBundle));
+
+        return parameters;
+    }
+
+    public MeasureReport createMeasureReport(String url) throws ClientProtocolException, IOException {
+
+        MeasureReport report = new MeasureReport();
+        report.setStatus(MeasureReport.MeasureReportStatus.COMPLETE);
+        report.setType(MeasureReport.MeasureReportType.DATACOLLECTION);
+        report.setMeasure(PlirConstants.MEASURE_RESOURCE_MEASURE);
+
+        Bundle bundle = fetchBundle(url);
+
+        List<Reference> references = new ArrayList<Reference>();
+
+        for (Bundle.BundleEntryComponent entry : bundle.getEntry()) {
+            if (entry.getResource() instanceof Observation) {
+                Observation resource = (Observation) entry.getResource();
+                Reference subjectRef = resource.getSubject();
+                Reference encounterRef = resource.getEncounter();
+                references.add(subjectRef);
+                references.add(encounterRef);
+            }
+        }
+        report.setEvaluatedResource(references);
+        return report;
+    }
+
+    public Bundle createObsBundle(String url) throws ClientProtocolException, IOException {
+        Bundle observationBundle = new Bundle();
+        Bundle bundle = fetchBundle(url);
+        observationBundle.setType(Bundle.BundleType.COLLECTION);
+
+        for (Bundle.BundleEntryComponent entry : bundle.getEntry()) {
+            if (entry.getResource() instanceof Observation) {
+                Resource resource = entry.getResource();
+                Bundle.BundleEntryComponent component = observationBundle.addEntry();
+                component.setResource(resource);
+            }
+        }
+        return observationBundle;
+    }
+
+    private String constructUrl(HttpServletRequest theServletRequest, Measure measure, String periodStart,
+            String periodEnd) {
+        StringBuffer baseUrl = new StringBuffer(theServletRequest.getScheme());
+        baseUrl.append("://");
+        baseUrl.append(theServletRequest.getServerName());
+        baseUrl.append(":");
+        baseUrl.append(theServletRequest.getServerPort());
+
+        String expression = measure.getGroupFirstRep().getPopulationFirstRep().getCriteria().getExpression();
+
+        StringBuffer params = new StringBuffer("&date=ge");
+        params.append(periodStart);
+        params.append("&date=lt");
+        params.append(periodEnd);
+
+        String fullUrl = baseUrl + "/fhir/" + expression + params;
```

Comment:
```suggestion
        String fullUrl = baseUrl + theServletRequest.getContext() + expression + params;
```

Reply:
@ibacher , atleast from running test, `getContext()` returns an empty String , more of like this is deployed at the ROOT.
so i had to hard code the string `"/fhir/"`

Reply:
Yeah, I guess the `context` is going to be `null` here. It would still be good if we could somehow derive this from the request itself, but it's not a big deal.

Reply:
its actually theServletRequest.getServletPath() , my bad